hoodie-client/src/main/java/com/uber/hoodie/index/bloom/HoodieBloomIndex.java

```diff
@@ -331,28 +330,32 @@ public boolean isImplicitWithStorage() {
       final Map<String, List<BloomIndexFileInfo>> partitionToFileIndexInfo,
       JavaPairRDD<String, String> partitionRecordKeyPairRDD, int shuffleParallelism, HoodieTableMetaClient metaClient,
       Map<String, Long> fileGroupToComparisons) {
-    JavaPairRDD<String, Tuple2<String, HoodieKey>> fileSortedTripletRDD =
+    JavaRDD<Tuple2<String, HoodieKey>> fileComparisonsRDD =
         explodeRecordRDDWithFileComparisons(partitionToFileIndexInfo, partitionRecordKeyPairRDD);
+
     if (config.useBloomIndexBucketizedChecking()) {
-      BucketizedBloomCheckPartitioner partitioner = new BucketizedBloomCheckPartitioner(shuffleParallelism,
-          fileGroupToComparisons, config.getBloomIndexKeysPerBucket());
-      fileSortedTripletRDD = fileSortedTripletRDD.repartitionAndSortWithinPartitions(partitioner);
+      Partitioner partitioner = new BucketizedBloomCheckPartitioner(
+          shuffleParallelism,
+          fileGroupToComparisons,
+          config.getBloomIndexKeysPerBucket()
+      );
+
+      fileComparisonsRDD = fileComparisonsRDD
+          .mapToPair(t -> new Tuple2<>(String.format("%s#%s", t._1, t._2.getRecordKey()), t))
```

Comment:
Because the Partitioner doesn't have access to the values of the PairRDD, I had to somehow include the record key in the key along with the file name. I was trying to find a way to support this without making a composite key here.

Reply:
ah. we would have to shuffle extra then, looks like.. could we use Tuple2 or a Pair again instead of `#` concatenation for the key.?

Reply:
I think `Pair` will work since it implements Comparable, I'll update